apps/engineeringblog/app/page.tsx

```diff
@@ -1,9 +1,23 @@
-import { Box, Container } from "@mui/material";
+import { Container } from "@mui/material";
+import { ArticleList } from "@/engineeringblog/components/Article";
+import { getAllPosts } from "@/engineeringblog/utils";
+import Empty from "@/engineeringblog/components/Empty";
+
+export default async function index() {
+  const posts = await getAllPosts();
+
+  if (!posts.length) {
+    return <Empty />;
+  }
 
-export default function index() {
   return (
-    <Container>
-      <Box minHeight="1000px">Homepage</Box>
+    <Container
```

Comment:
@kilemensi Because that page is server-rendered. Using any component from `commons-ui` in any server-rendered page or component throws an error about client-side components being used server-side. The common issue I have encountered is with `RichTypography` which has a `useEffect` which can only be used client-side. It is not used in `Section` but affects all `commons-ui` components if used in server-rendered components.

Reply:
@kilemensi I have added #878 to track this.